org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java

```diff
@@ -1738,10 +1754,16 @@ protected ITreeAppendable toLambda(XClosure closure, ITreeAppendable b, Lightwei
 			b.openPseudoScope();
 			b.append("(");
 			List<JvmFormalParameter> closureParams = closure.getFormalParameters();
+			boolean isVarArgs = operation.isVarArgs();
 			for (int i = 0; i < closureParams.size(); i++) {
 				JvmFormalParameter closureParam = closureParams.get(i);
 				LightweightTypeReference parameterType = getClosureOperationParameterType(type, operation, i);
-				b.append(parameterType);
+				if (isVarArgs && i == closureParams.size()-1 && parameterType instanceof ArrayTypeReference) {
+					b.append(((ArrayTypeReference) parameterType).getComponentType());
```

Comment:
Minor: The cast is not necessary. All LightweightTypeRefs have a method getComponentType. Also isArray favorable over instanceof ArrayTypeReference since it encapsulates implementation details, e.g. for multi type refs etc.